CMakeLists.txt

```diff
@@ -325,9 +325,15 @@ if(AVIF_ENABLE_NODISCARD)
     else()
         unset(CMAKE_C_STANDARD)
         set_property(TARGET avif_obj PROPERTY C_STANDARD)
-        target_compile_options(avif_obj PUBLIC $<BUILD_INTERFACE:$<$<COMPILE_LANGUAGE:C>:-std=gnu2x>>)
+        if(CMAKE_C_COMPILER_ID MATCHES "Clang" OR CMAKE_C_COMPILER_ID MATCHES "GNU")
```

Comment:
Can also be written as:
```CMake
target_compile_options(avif_obj PUBLIC 
     $<$<OR:$<CXX_COMPILER_ID:Clang>,$<CXX_COMPILER_ID:AppleClang>,$<CXX_COMPILER_ID:GNU>>:
          -std=gnu2x>
     $<$<CXX_COMPILER_ID:MSVC>:
          /std:clatest>)
```

Reply:
Thanks for the suggestion. This is harder to read. Also it omits the `$<BUILD_INTERFACE:$<$<COMPILE_LANGUAGE:C>` part used in the current code. Is that not neeeded?